opstack/src/builder.rs

```diff
@@ -18,6 +19,7 @@ pub struct OpStackClientBuilder {
     consensus_rpc: Option<Url>,
     execution_rpc: Option<Url>,
     rpc_socket: Option<SocketAddr>,
+    eth_network: Option<EthNetwork>,
```

Comment:
Oh here I had meant we should support passing in the opstack `Network` which we can use to fill out the `ChainConfig` values that the builder needs, but good catch that we also need to know the Ethereum network that it is built on. Maybe we can add `eth_network` as a parameter to the opstack `ChainConfig` and then just allow setting the opstack `Network` with a `network` method on the builder, which is then used to grab the correct `ChainConfig` so that we don't need to set things like the chain id, system config address, and default unsafe signer manually.

Reply:
Yeah that's cleaner